addons/website/models/ir_ui_view.py

```diff
@@ -93,6 +93,44 @@ def write(self, vals):
 
         return True
 
+    @api.multi
+    def _get_copied_views(self):
+        """ Given a generic view (no website_id), return a record set containing
+            all the views that have been COW'd from it.
+        """
+        self.ensure_one()
+        domain = [('key', '=', self.key), ('website_id', '!=', None)]
+        return self.with_context(active_test=False).search(domain)
+
+    def _hook_write(self, datas):
+        if self.type == 'qweb' and self.inherit_id and not self.website_id and not self.inherit_id.website_id:
```

Comment:
The 'has to be an inherited view' makes sense for the creation as we don't need/want to COW a view if it is a top level one (it will be using that view on a website context)
But for the update, so this line, I think we can drop the part `self.inherit_id` and `self.inherit_id.website_id`